lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -93,6 +94,38 @@ Var JavascriptObject::EntryHasOwnProperty(RecyclableObject* function, CallInfo c
     JIT_HELPER_END(Object_HasOwnProperty);
 }
 
+Var JavascriptObject::EntryHasOwn(RecyclableObject* function, CallInfo callInfo, ...)
+{
+    JIT_HELPER_REENTRANT_HEADER(Object_HasOwn);
+    PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+
+    ARGUMENTS(args, callInfo);
+    ScriptContext* scriptContext = function->GetScriptContext();
+
+    Assert(!(callInfo.Flags & CallFlags_New));
+
+    RecyclableObject* dynamicObject = nullptr;
+    if (args.Info.Count < 2 || FALSE == JavascriptConversion::ToObject(args[1], scriptContext, &dynamicObject))
+    {
+        JavascriptError::ThrowTypeError(scriptContext, JSERR_FunctionArgument_NeedObject, _u("Object.hasOwn"));
+    }
+
+    Var propertyName = args.Info.Count == 2 ? scriptContext->GetLibrary()->GetUndefined() : args[2];
```

Comment:
I was wondering what happens if count is greater than 3, but then realized it is limited to 3.

Reply:
It can be larger, but it's irrelevant, cases:
0 : some kind of internal error/can't happen
1 : called with no arguments (just has a 'this' value) - throw type error above
2 : called with 1 argument (the object to look at) - use undefined as the property we're looking for
3 : called with 2 arguments - look for 2nd argument as the target property
4+ : treated like 3, extra arguments are ignored